modules/Hoot/tools/grailPull.js

```diff
@@ -114,6 +114,14 @@ export default class GrailPull {
             .then( () => Hoot.folders.refreshAll() )
             .then( () => Hoot.events.emit( 'render-dataset-table' ) );
 
+
+        let history = JSON.parse( Hoot.context.storage('history') );
+        if ( history.bboxHistory.length === 5 ) {
```

Comment:
I do not think anyone would have be clever and mutate the history to have > 5 entries, but they could. 
So should we do a greater than or equal to? 
And then instead of pop do a slice like `history.bboxHistory = history.bboxHistory.slice(1,5)`.

This ensures in the greater than or equal scenario we reset to first 4 bboxes before we add the new bbox on line 122

Reply:
true. better to be safe